modules/__tests__/LocationUtils-test.js

```diff
@@ -111,4 +111,123 @@ describe('createLocation', () => {
       })
     })
   })
+
+  describe('with a current location', () => {
+    const currentLocation = {
+      pathname: '/The/Old/Path',
+      search: '?theQuery',
+      hash: '#fragment'
+    };
+
+    const createRelative = url => createLocation(url, undefined, undefined, currentLocation)
+
+    describe('can resolve a relative pathname', () => {
+      describe('given as a string', () => {
+        it('has the correct properties', () => {
+          expect(createRelative('new/path')).toMatch({
+            pathname: '/The/Old/new/path',
+            search: '',
+            hash: ''
+          })
+        })
+      })
+      describe('given as an object', () => {
+        it('has the correct properties', () => {
+          expect(createRelative({ pathname: 'new/path' })).toMatch({
```

Comment:
Ah, good call to follow existing style. I'd like to slim down if we can and avoid duplication. You can stick with just the more convenient string form in these tests.